services/hockeyapp/hockeyapp.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('joi')
+const { renderVersionBadge } = require('../../lib/version')
+const BaseJsonService = require('../base-json')
+const { NotFound } = require('../errors')
+
+const hockeyappSchema = Joi.object({
+  app_versions: Joi.array().items(
+    Joi.object({
+      version: Joi.string().required(),
+      shortversion: Joi.string().required(),
+    })
+  ),
+}).required()
+
+module.exports = class Hockeyapp extends BaseJsonService {
+  async fetch({ apptoken, appid }) {
+    const url = `https://rink.hockeyapp.net/api/2/apps/${appid}/app_versions`
+    const options = { headers: { 'X-HockeyAppToken': apptoken } }
+    return this._requestJson({
+      url,
+      schema: hockeyappSchema,
+      options: options,
+    })
+  }
+
+  static render({ version }) {
+    return renderVersionBadge({ version })
+  }
+
+  async handle({ apptoken, appid }) {
+    const json = await this.fetch({ apptoken, appid })
+
+    if (json.app_versions === undefined || json.app_versions.length === 0) {
```

Comment:
Your schema says that the `app_versions` key must be an array.
How does the situation where `json.app_versions === undefined` occur?
Also, given there are some custom criteria here for the 'not found' response, can we have service tests covering these cases please?

Reply:
The undefined check is just a sanity check but as it will be checked by Joi anyways I will remove the undefined check and rely on that. Also adding the test case you mentioned.